fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergRewriteDataJob.java

```diff
@@ -85,33 +151,121 @@ public void prepare() throws Exception {
+                    ExecPlan localPlan = StatementPlanner.plan(parsedStmt, subCtx);
+    
+                    List<IcebergScanNode> localScanNodes = localPlan.getFragments().stream()
+                            .flatMap(f -> f.collectScanNodes().values().stream())
+                            .filter(s -> s instanceof IcebergScanNode && "IcebergScanNode".equals(s.getPlanNodeName()))
+                            .map(s -> (IcebergScanNode) s)
+                            .collect(Collectors.toList());
+    
+                    if (localScanNodes.isEmpty()) {
+                        LOG.info("No IcebergScanNode in sub plan. Skip one task group.");
+                        return null;
+                    }
+                    for (IcebergScanNode sn : localScanNodes) {
+                        sn.rebuildScanRange(res);
+                    }
+    
+                    StmtExecutor exec = StmtExecutor.newInternalExecutor(subCtx, localStmt);
+                    if (context.getExecutor() != null) {
+                        context.getExecutor().registerSubStmtExecutor(exec);
+                    }
+                    try {
+                        exec.handleDMLStmt(localPlan, localStmt);
+                    } finally {
+                        exec.addFinishedQueryDetail();
+                    }
+                    return null;
+                }));
+            }
+    
+            for (Future<?> future : futures) {
+                try {
+                    future.get();
+                } catch (Exception e) {
+                    LOG.error("Failed to compact files on [{}]", originAlterStmt.getTableName(), e);
+                    throw new StarRocksConnectorException("Failed to compact files", e);
                 }
-                context.setQueryId(UUIDUtil.genUUID());
-                context.setExecutionId(UUIDUtil.toTUniqueId(context.getQueryId()));
-                executor.handleDMLStmt(execPlan, rewriteStmt);
+            }
+            IcebergSinkExtra extra = new IcebergSinkExtra();
+            List<TSinkCommitInfo> faList = Lists.newArrayList();
+            for (FinishArgs fa : collected) {
+                extra.addScannedDataFiles(((IcebergSinkExtra) fa.getExtra()).getScannedDataFiles());
+                extra.addAppliedDeleteFiles(((IcebergSinkExtra) fa.getExtra()).getAppliedDeleteFiles());
+                faList.addAll(fa.getCommitInfos());
+            }
+            try {
+                context.getGlobalStateMgr().getMetadataMgr().finishSink(
+                        collected.peek().getCatalog(), collected.peek().getDb(), collected.peek().getTable(),
+                        faList, collected.peek().getBranch(), extra);
+            } catch (Exception e) {
+                LOG.error("Failed to commit iceberg rewrite on [{}]", originAlterStmt.getTableName(), e);
+                throw new StarRocksConnectorException("Failed to commit iceberg rewrite", e);
```

Comment:
all these exception you catched, do not clean the state, like the files , is this right?